src/System.Reflection.Metadata/tests/Metadata/MetadataReaderTests.cs

```diff
@@ -92,8 +92,202 @@ internal static unsafe GCHandle GetPinnedPEImage(byte[] peImage)
             return pinned;
         }
 
+        internal static unsafe int FindIndex(byte[] peImage, byte[] toFind, int start)
+        {
+            byte[] toTest = new byte[toFind.Length];
+            for (int i = 0; i < peImage.Length - toFind.Length; i++)
+            {
+                Array.Copy(peImage, i + start, toTest, 0, toTest.Length);
+                if (toTest.SequenceEqual(toFind))
```

Comment:
>SequenceEqual [](start = 27, length = 13)

Don't we have SequenceEqual somewhere that takes a start index and length, so that we don't need to copy the bytes?

Reply:
I can't find a SequenceEqual that takes a start index or a length. I changed to using Slice inside of SequenceEqual and it performs better.